docs/process/process_areas/implementation/guidance/python/coding_guidelines.rst

```diff
@@ -0,0 +1,89 @@
+
+
+**Consistency within score matters more than personal preference**.
+
+The art lies in balancing completeness of this document vs keeping it short, since frankly, no one
+will read a long document. Let's focus on issues that actively come up in real life and are worth
+discussing and agreeing upon, not on things that *might* be important. (We don't want to write a
+book here. And we don't have to, since Python is not used for safety critical systems.)
+
+This guide is a **living document**—keep it concise, relevant, and practical.
+
+We assume basic programming and software-engineering know-how and will not repeat basics here! Go
+ask Google or ChatGPT. Or one of these established Python best practice websites:
+
+- [PEP 8](https://pep8.org/)
+- [The Hitchhiker's Guide to Python](https://docs.python-guide.org/)
+- [Real Python](https://realpython.com/)
+
+Comments
+--------
+- The first rule of comments is to write them **only when something is worth explaining**.
+- Comments should explain *why* the code does something, not *what* it does.
+- Avoid redundant comments: if the code is clear enough, a comment is not needed.
+
+Testing
+-------
+- **Readability:** For some reason, people tend to forget all they have ever learned about
+  readability when writing tests. **Tests do not have tests.** They must be even more readable than
+  production code!
+- **Mocking is an anti-pattern.** Well... it's complex. We don't want to write a book on mocking
+  here. Just really, really consider whether mocking is the best approach. - Favor dependency
+  injection and lightweight fixtures over excessive mocking. Mocks should not replace proper
+  design.
+- **Do not test every single function or class!** Instead: - Focus on testing *behavior*, not
+  implementation details. Overly fine-grained tests lead to brittle code and high maintenance.
+- **Boundary conditions:** - Edge cases often break things. Think about invalid input, large data
+  sets, empty cases, and concurrency issues.
+
+Documentation
+--------------------
+- If your documentation doesn't fit into a comment, create a `README.md` in the directory of the
+  tool.
+- `README.md` should explain the overall architecture and key design decisions.
+- If the design impacts multiple components, consider an architecture diagram.
+- User-facing documentation is stored under `docs/guidance`.
+
+
+Tools & Versions
+================
+
+We're using Python, Ruff (linting & formatting), Pyright (linting) and pytest for our tooling. The
+versions are managed by the infrastructure community and will be updated as frequently as possible.
+The configuration is managed by infrastructure and process communities.
+
+*Compliance to all tools is mandatory.*
+
+Currently we do NOT aim at supporting older Python versions.
+
+Templates, Checklists and other guidelines
+------------------------------------------------
+
+While the S-CORE templates and checklists are far from useless, and should be adhered to in
```

Comment:
"are far from useless" is double negative and indicate judgement that someone could find them useful. Better state something like...

 ```suggestion
The S-CORE architecture and implementation templates and inspection checklists are helpful, and should be adhered to, although in
```

Reply:
Guidelines moved to contribute as discussed with @hoe-jo, therefore any reference to checklists etc is no longer part of this PR. This is only coding guidelines.